boltstub_rs/src/values/bolt_struct/unknown_type.rs

```diff
@@ -0,0 +1,140 @@
+use std::fmt::{Debug, Display, Formatter};
+
+use serde_json::Value as JsonValue;
+
+use super::_parsing::{check_last_json_field, next_json_field, next_pack_stream_field};
+use crate::bolt_version::JoltVersion;
+use crate::parse_error::ParseError;
+use crate::parser::ActorConfig;
+use crate::values::bolt_struct::TAG_UNKNOWN_TYPE;
+use crate::values::pack_stream_value::{PackStreamStruct, PackStreamValue};
+use indexmap::IndexMap;
+
+#[derive(Debug, Clone)]
+pub(crate) struct JoltUnknownType {
+    pub(crate) name: String,
+    pub(crate) minimum_protocol_major: i64,
+    pub(crate) minimum_protocol_minor: i64,
+    pub(crate) message: Option<String>,
+}
+
+impl JoltUnknownType {
+    pub(crate) fn parse(
+        v: JsonValue,
+        jolt_version: JoltVersion,
+        config: &ActorConfig,
+    ) -> Result<Self, ParseError> {
+        if !matches!(jolt_version, JoltVersion::V3) {
+            return Err(ParseError::new(format!(
+                "Sigil \"UT\" (unknown type) can only be parsed in JoltVersion::V3, using {jolt_version:?}"
+            )));
+        }
+        let JsonValue::Array(fields) = v else {
+            return Err(ParseError::new(
+                "Expected array after sigil \"UT\", but found {v:?}",
+            ));
+        };
+        let mut fields = fields.into_iter().enumerate();
+        let i = 0;
+
+        let (i, name) = next_json_field::<String>(&mut fields, "name", i, "UT", config)?;
+        let (i, minimum_protocol_major) =
+            next_json_field::<i64>(&mut fields, "minimum_protocol_major", i, "UT", config)?;
+        let (i, minimum_protocol_minor) =
+            next_json_field::<i64>(&mut fields, "minimum_protocol_minor", i, "UT", config)?;
+        let (i, extra) = next_json_field::<IndexMap<String, PackStreamValue>>(
+            &mut fields,
+            "extra",
+            i,
+            "UT",
+            config,
+        )?;
+        check_last_json_field(&mut fields, i, "UT")?;
+        if extra.contains_key("message") {
+            let PackStreamValue::String(ref message) = extra["message"] else {
+                return Err(ParseError::new("Expected message in extra to be string."));
+            };
+            Ok(Self {
+                name,
+                minimum_protocol_major,
+                minimum_protocol_minor,
+                message: Some(message.clone()),
+            })
+        } else {
+            Ok(Self {
+                name,
+                minimum_protocol_major,
+                minimum_protocol_minor,
+                message: None,
+            })
+        }
+    }
+
+    pub(crate) fn into_struct(self) -> PackStreamStruct {
+        let mut extra_map = IndexMap::with_capacity(1);
+        if self.message.is_some() {
+            extra_map.insert(
+                String::from("message"),
+                PackStreamValue::String(self.message.unwrap()),
+            );
+        }
```

Comment:
```suggestion
        if let Some(message) = self.message {
            extra_map.insert(String::from("message"), PackStreamValue::String(message));
        }
```
100% panic-path-free[^1] code :relieved: 

[^1]: bearing OOMs and other fatal things.